src/ORM/FieldType/DBForeignKey.php

```diff
@@ -22,40 +24,26 @@
  */
 class DBForeignKey extends DBInt
 {
-    /**
-     * @var DataObject
-     */
-    protected $object;
+    protected ?DataObject $object;
```

Comment:
Explicitly `DataObject` rather than `ViewableData` here unlike other `DBField` implementations, because the foreignkey represents a relation to a `DataObject`.